cms/djangoapps/contentstore/views/transcripts_ajax.py

```diff
@@ -643,17 +705,26 @@ def replace_transcripts(request):
 def _get_item(request, data):
     """
     Obtains from 'data' the locator for an item.
-    Next, gets that item from the modulestore (allowing any errors to raise up).
+    Next, gets that item from the modulestore (allowing any errors to raise up)
+    or from library API if is a library content.
     Finally, verifies that the user has access to the item.
 
-    Returns the item.
+    Returns the item and a boolean if is a library content.
     """
     usage_key = UsageKey.from_string(data.get('locator'))
-    if not usage_key.context_key.is_course:
-        # TODO: implement transcript support for learning core / content libraries.
-        raise TranscriptsRequestValidationException(_('Transcripts are not yet supported in content libraries.'))
+
+    context_key = usage_key.context_key
+    if not context_key.is_course:
+        if isinstance(context_key, LibraryLocatorV2):
+            lib_api.require_permission_for_library_key(
+                context_key,
+                request.user,
+                lib_api.lib_permissions.CAN_EDIT_THIS_CONTENT_LIBRARY
+            )
+            return xblock_api.load_block(usage_key, request.user)
```

Comment:
There is already a public API for this:
```suggestion
            return xblock_api.load_block(usage_key, request.user, check_permission=CheckPerm.CAN_EDIT)
```

Reply:
Updated: https://github.com/openedx/edx-platform/pull/36058/commits/5bb426f505c1f0ba3b862f992a62dd2ac78091bd